Move UserMediaCaptureManagerProxy to WebKit/GPUProcess

youennf opened the pull request:
#### c46feb49175610e228377933235f5cac40710ba1
<pre>
Move UserMediaCaptureManagerProxy to WebKit/GPUProcess
<a href="https://rdar.apple.com/155411315">rdar://155411315</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=295627">https://bugs.webkit.org/show_bug.cgi?id=295627</a>

Reviewed by Anne Van Kesteren.

We move UserMediaCaptureManagerProxy files to WebKit/GPUPProcess since they are only used in GPUProcess.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.cpp: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp.
(WebKit::UserMediaCaptureManagerProxy::create):
(WebKit::UserMediaCaptureManagerProxy::UserMediaCaptureManagerProxy):
(WebKit::UserMediaCaptureManagerProxy::~UserMediaCaptureManagerProxy):
(WebKit::UserMediaCaptureManagerProxy::createMicrophoneSource):
(WebKit::canCaptureFromMultipleCameras):
(WebKit::UserMediaCaptureManagerProxy::createCameraSource):
(WebKit::UserMediaCaptureManagerProxy::createMediaSourceForCaptureDeviceWithConstraints):
(WebKit::UserMediaCaptureManagerProxy::startProducingData):
(WebKit::UserMediaCaptureManagerProxy::stopProducingData):
(WebKit::UserMediaCaptureManagerProxy::removeSource):
(WebKit::UserMediaCaptureManagerProxy::capabilities):
(WebKit::UserMediaCaptureManagerProxy::applyConstraints):
(WebKit::UserMediaCaptureManagerProxy::clone):
(WebKit::UserMediaCaptureManagerProxy::takePhoto):
(WebKit::UserMediaCaptureManagerProxy::getPhotoCapabilities):
(WebKit::UserMediaCaptureManagerProxy::getPhotoSettings):
(WebKit::UserMediaCaptureManagerProxy::endProducingData):
(WebKit::UserMediaCaptureManagerProxy::setShouldApplyRotation):
(WebKit::UserMediaCaptureManagerProxy::setIsInBackground):
(WebKit::UserMediaCaptureManagerProxy::isPowerEfficient):
(WebKit::UserMediaCaptureManagerProxy::clear):
(WebKit::UserMediaCaptureManagerProxy::close):
(WebKit::UserMediaCaptureManagerProxy::setOrientation):
(WebKit::UserMediaCaptureManagerProxy::rotationAngleForCaptureDeviceChanged):
(WebKit::UserMediaCaptureManagerProxy::hasSourceProxies const):
(WebKit::UserMediaCaptureManagerProxy::sharedPreferencesForWebProcess const):
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.h: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.h.
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::protectedConnection):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::protectedLogger):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setCaptureAttributionString):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setTCCIdentity):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setCurrentMediaEnvironment):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::startProducingData):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::remoteVideoFrameObjectHeap):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::startMonitoringCaptureDeviceRotation):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::stopMonitoringCaptureDeviceRotation):
(WebKit::UserMediaCaptureManagerProxy::didReceiveMessageFromGPUProcess):
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.messages.in: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.messages.in.
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/UserMediaPermissionRequestManagerProxy.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/297203@main">https://commits.webkit.org/297203@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/936e0f4a8df0f70800529f7b4c9a36261ba0fa71

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110884 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30547 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20982 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116914 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61152 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31227 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39133 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84302 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113832 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24946 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99838 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64745 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24298 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60712 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94331 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18042 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119716 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37929 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28184 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93259 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38305 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96112 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93084 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23719 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38135 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15878 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33904 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37822 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43293 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37485 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40821 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39190 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -2589,7 +2588,6 @@
 		F4E090952D889D6A00322226 /* WKIdentityDocumentPresentmentDelegate.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E090942D889CB800322226 /* WKIdentityDocumentPresentmentDelegate.h */; };
 		F4E090992D88AAC300322226 /* WKIdentityDocumentPresentmentRawRequest.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E090962D88A66F00322226 /* WKIdentityDocumentPresentmentRawRequest.h */; };
 		F4E28A362C923814008120DD /* ScriptTrackingPrivacyFilter.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E28A352C923814008120DD /* ScriptTrackingPrivacyFilter.h */; };
-		F4E44EB72DCD33E800A304C4 /* ISO18013MobileDocumentRequest+Extras.swift in Sources */ = {isa = PBXBuildFile; fileRef = F4E44EB62DCD33E800A304C4 /* ISO18013MobileDocumentRequest+Extras.swift */; };
```
This seems unrelated.

annevk commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -8005,7 +8003,7 @@
 		CD8252E025D4918400862FD8 /* RemoteRemoteCommandListenerMessageReceiver.cpp */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.cpp.cpp; path = RemoteRemoteCommandListenerMessageReceiver.cpp; sourceTree = "<group>"; };
 		CD8252E125D4918500862FD8 /* RemoteRemoteCommandListenerMessages.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = RemoteRemoteCommandListenerMessages.h; sourceTree = "<group>"; };
-		CD95493526159004008372D9 /* .dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = .dylib; sourceTree = BUILT_PRODUCTS_DIR; };
+		CD95493526159004008372D9 /* libWebKitSwift.dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = libWebKitSwift.dylib; sourceTree = BUILT_PRODUCTS_DIR; };
```
This seems unrelated?

annevk commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -8890,7 +8888,7 @@
 				5CE4B62029CF877F0038F565 /* GPUExtension.appex */,
 				7B9FC59D28A28F88007570E7 /* libWebKitPlatform.a */,
-				CD95493526159004008372D9 /* .dylib */,
+				CD95493526159004008372D9 /* libWebKitSwift.dylib */,
```
Unrelated?

annevk commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -19026,7 +19023,7 @@
 			name = WebKitSwift;
 			productName = WebKitSwift;
-			productReference = CD95493526159004008372D9 /* .dylib */;
+			productReference = CD95493526159004008372D9 /* libWebKitSwift.dylib */;
```
Unrelated?

annevk commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -21089,7 +21086,6 @@
 				0735F31E2D3760BC0003D029 /* IntelligenceTextEffectChunk.swift in Sources */,
 				0735F31C2D3760AA0003D029 /* IntelligenceTextEffectViewManager.swift in Sources */,
 				E838FCB02DE90BF800703353 /* ISO18013MobileDocumentRequest+Extras.swift in Sources */,
-				A1098E6D2BB6436400449EE0 /* LinearMediaKitExtras.swift in Sources */,
```
Unrelated?

youennf replied:
Hum, Xcode did it for me, I'll remove those changes.

youennf committed:
Move UserMediaCaptureManagerProxy to WebKit/GPUProcess
rdar://155411315
https://bugs.webkit.org/show_bug.cgi?id=295627

Reviewed by Anne Van Kesteren.

We move UserMediaCaptureManagerProxy files to WebKit/GPUPProcess since they are only used in GPUProcess.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.cpp: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp.
(WebKit::UserMediaCaptureManagerProxy::create):
(WebKit::UserMediaCaptureManagerProxy::UserMediaCaptureManagerProxy):
(WebKit::UserMediaCaptureManagerProxy::~UserMediaCaptureManagerProxy):
(WebKit::UserMediaCaptureManagerProxy::createMicrophoneSource):
(WebKit::canCaptureFromMultipleCameras):
(WebKit::UserMediaCaptureManagerProxy::createCameraSource):
(WebKit::UserMediaCaptureManagerProxy::createMediaSourceForCaptureDeviceWithConstraints):
(WebKit::UserMediaCaptureManagerProxy::startProducingData):
(WebKit::UserMediaCaptureManagerProxy::stopProducingData):
(WebKit::UserMediaCaptureManagerProxy::removeSource):
(WebKit::UserMediaCaptureManagerProxy::capabilities):
(WebKit::UserMediaCaptureManagerProxy::applyConstraints):
(WebKit::UserMediaCaptureManagerProxy::clone):
(WebKit::UserMediaCaptureManagerProxy::takePhoto):
(WebKit::UserMediaCaptureManagerProxy::getPhotoCapabilities):
(WebKit::UserMediaCaptureManagerProxy::getPhotoSettings):
(WebKit::UserMediaCaptureManagerProxy::endProducingData):
(WebKit::UserMediaCaptureManagerProxy::setShouldApplyRotation):
(WebKit::UserMediaCaptureManagerProxy::setIsInBackground):
(WebKit::UserMediaCaptureManagerProxy::isPowerEfficient):
(WebKit::UserMediaCaptureManagerProxy::clear):
(WebKit::UserMediaCaptureManagerProxy::close):
(WebKit::UserMediaCaptureManagerProxy::setOrientation):
(WebKit::UserMediaCaptureManagerProxy::rotationAngleForCaptureDeviceChanged):
(WebKit::UserMediaCaptureManagerProxy::hasSourceProxies const):
(WebKit::UserMediaCaptureManagerProxy::sharedPreferencesForWebProcess const):
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.h: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.h.
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::protectedConnection):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::protectedLogger):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setCaptureAttributionString):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setTCCIdentity):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::setCurrentMediaEnvironment):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::startProducingData):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::remoteVideoFrameObjectHeap):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::startMonitoringCaptureDeviceRotation):
(WebKit::UserMediaCaptureManagerProxy::ConnectionProxy::stopMonitoringCaptureDeviceRotation):
(WebKit::UserMediaCaptureManagerProxy::didReceiveMessageFromGPUProcess):
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.messages.in: Renamed from Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.messages.in.
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/UserMediaPermissionRequestManagerProxy.mm:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/297203@main
